### PR TITLE
Create encrypted /srv bound with clevis to the TPM2

### PR DESCRIPTION
**Pull request by haraldh**
On reinstall work around anaconda and don't format /srv and /home.
Also reconstruct missing /etc/crypttab and /etc/fstab for /srv.

Install clevis systemd units to open a luks in local-fs.target.
It does not need network and remote-fs.
Also we want to have it ready before any other daemons start,
in case we have the configuration in /srv.

### CODE REVIEW

File: enarx.ks [hunk 1]
```diff
@@ -90,10 +90,88 @@ Type=ether
...
+else
+
+   # umount /srv and close the luks device
+   if umount /srv &>/dev/null || ! mountpoint /srv &>/dev/null; then
+      cryptsetup close /dev/disk/by-label/srv &>/dev/null || :
+   fi
+
+   # bind /srv to the tpm
+   if ! clevis luks unlock -d "$luksdev"; then
+      clevis luks bind -f -k- -d "$luksdev" tpm2 '{}' <<< "temppass"
```
Comment (npmccallum) [inline enarx.ks:135]: Are we sure this uses the right PCR?

File: enarx.ks [hunk 2]
```diff
@@ -90,10 +90,88 @@ Type=ether
...
+[Unit]
+Description=Local Clevis LUKS systemd-ask-password Responder
+DefaultDependencies=no
+
+[Service]
+Type=oneshot
+ExecStart=/usr/libexec/clevis-luks-askpass -l
+EOF
+
+systemctl enable local-clevis-luks-askpass.path
```
Comment (npmccallum) [inline enarx.ks:168]: I don't understand why this unit is needed.

File: enarx.ks [hunk 3]
```diff
@@ -90,10 +90,88 @@ Type=ether
...
+
+# give the /srv luks partition a partition label
+if ! [[ -b /dev/disk/by-partlabel/srv ]]; then
+   luksdev=/dev/$(ls -1 /sys/dev/block/$(get_maj_min /dev/disk/by-label/srv)/slaves)
+
+   # label the srv partition for later
+   sfdisk --part-label \
+       /dev/block/$(cat /sys/dev/block/$(get_maj_min $luksdev)/../dev) \
+       $(cat /sys/dev/block/$(get_maj_min "$luksdev")/partition) \
+       srv
```
Comment (npmccallum) [inline enarx.ks:108]: I don't understand why this branch is needed.

File: enarx.ks (hunk 2 above)
Comment (haraldh) [reply enarx.ks:168]: The original clevis-luks-askpass.path is for `remote-fs.target` only.

File: enarx.ks (hunk 3 above)
Comment (haraldh) [reply enarx.ks:108]: I need `/dev/disk/by-partlabel/srv` to specify 
```
part /srv  --label=srv  --onpart=disk/by-partlabel/srv --encrypted --noformat --passphrase=temppass
```
on reinstalls... note the `--onpart=disk/by-partlabel/srv`

File: enarx.ks (hunk 1 above)
Comment (haraldh) [reply enarx.ks:135]: I don't know.. do we need to specify a PCR?

File: enarx.ks (hunk 2 above)
Comment (haraldh) [reply enarx.ks:168]: and we need
```
DefaultDependencies=no
```

File: enarx.ks (hunk 1 above)
Comment (npmccallum) [reply enarx.ks:135]: We probably should be explicit here. Otherwise we might end up with unrecoverable data.

File: enarx.ks (hunk 2 above)
Comment (npmccallum) [reply enarx.ks:168]: But the only difference is the `netdev` option in `fstab` and `crypttab`, I'm fairly certain.

Comment (npmccallum) [reply enarx.ks:168]: i.e. Can't you just add `_netdev`?

Comment (haraldh) [reply enarx.ks:168]: Well, we want to bind to /etc as soon as possible before any daemons are started.. right?
Waiting for remote-fs.target might be too late.

**Comment by npmccallum** [comment]
We've decided to abandon this approach for a more simpler architecture. See #7.

**Pull request closed without merging**